### PR TITLE
bitget: ws subscribe stop orders channel 'ordersAlgo'

### PR DESCRIPTION
**Pull request by dmunozfer**
BITGET: subscribe to websockets stop orders channel 'ordersAlgo'

Bitget docs https://bitgetlimited.github.io/apidoc/en/mix/#plan-order-channel

### CODE REVIEW

**Comment by carlosmiei** [comment]
Hello @dmunozfer thanks for your contribution, we will review it soon!

File: js/src/pro/bitget.js [hunk 1]
```diff
@@ -588,6 +588,7 @@ export default class bitget extends bitgetRest {
         }
         let type = undefined;
         [type, params] = this.handleMarketTypeAndParams('watchOrders', market, params);
+        const isStop = this.safeValue(params, 'stop', false);
```
Comment (carlosmiei) [inline js/src/pro/bitget.js:591]: @dmunozfer I will update it no worries, but next time please change the correspondent typescript (.ts) file instead, the JS version is automatically compiled by tsc.

File: ts/src/pro/bitget.ts [hunk 2]
```diff
@@ -595,6 +595,7 @@ export default class bitget extends bitgetRest {
             marketId = market['id'];
             messageHash = messageHash + ':' + symbol;
         }
+        const isStop = this.safeValue (params, 'stop', false);
```
Comment (carlosmiei) [inline ts/src/pro/bitget.ts:598]: @dmunozfer also, next time we add a parameter like this we have to omit it from `params`

File: ts/src/pro/bitget.ts [hunk 3]
```diff
@@ -613,9 +614,10 @@ export default class bitget extends bitgetRest {
             }
         }
         const instId = (type === 'spot') ? marketId : 'default'; // different from other streams here the 'rest' id is required for spot markets, contract markets require default here
+        const channel = isStop ? 'ordersAlgo' : 'orders';
```
Comment (carlosmiei) [inline ts/src/pro/bitget.ts:617]: also for some reason this is not working:
```
 sending {
  op: 'subscribe',
  args: [
    { instType: 'spbl', channel: 'ordersAlgo', instId: 'LTCUSDT_SPBL' }
  ]
}
2023-05-14T10:08:43.716Z onMessage {
  event: 'error',
  code: 30001,
  msg: "instType:spbl,channel:ordersAlgo,instId:LTCUSDT_SPBL doesn't exist"
}

```

Comment (carlosmiei) [reply ts/src/pro/bitget.ts:617]: nvm only works with swap markets

File: ts/src/pro/bitget.ts [hunk 4]
```diff
@@ -613,9 +614,10 @@ export default class bitget extends bitgetRest {
...
         const args = {
             'instType': instType,
-            'channel': 'orders',
+            'channel': channel,
```
Comment (carlosmiei) [inline ts/src/pro/bitget.ts:620]: also it seems that stop orders updates are not correctly handled 
```
 onMessage {
  action: 'snapshot',
  arg: { instType: 'umcbl', channel: 'ordersAlgo', instId: 'default' },
  data: [
    {
      actualPx: '50.000000000',
      actualSz: '0.000000000',
      cOid: '1041586673338097665',
      cTime: '1684059535345',
      eps: 'api',
      hM: 'double_hold',
      id: '1041586673338097666',
      instId: 'LTCUSDT_UMCBL',
      key: '1041586673338097666',
      ordPx: '55.000000000',
      ordType: 'limit',
      planType: 'pl',
      posSide: 'long',
      side: 'buy',
      state: 'not_trigger',
      sz: '0.100000000',
      tS: 'open_long',
      tgtCcy: 'USDT',
      triggerPx: '55.000000000',
      triggerPxType: 'mark',
      triggerTime: '1684059535345',
      userId: '3704614084',
      version: 1041586673291960300
    }
  ]
}
```
usually when changing the subscription we also have to check `handleOrder/handleUpdate`

**Comment by carlosmiei** [comment]
DEMO:
```
✗ p bitget watchOrders None  None None '{"stop":true}' --swap      
Python v3.10.9
CCXT v3.0.102
bitget.watchOrders(None,None,None,{'stop': True})
[{'amount': 0.1,
  'average': None,
  'clientOrderId': '1041590898185019392',
  'cost': None,
  'datetime': '2023-05-14T10:35:42.627Z',
  'fee': None,
  'fees': [],
  'filled': None,
  'id': '1041590898185019393',
  'info': {'actualPx': '50.000000000',
           'actualSz': '0.000000000',
           'cOid': '1041590898185019392',
           'cTime': '1684060542627',
           'eps': 'api',
           'hM': 'double_hold',
           'id': '1041590898185019393',
           'instId': 'LTCUSDT_UMCBL',
           'key': '1041590898185019393',
           'ordPx': '55.000000000',
           'ordType': 'limit',
           'planType': 'pl',
           'posSide': 'long',
           'side': 'buy',
           'state': 'not_trigger',
           'sz': '0.100000000',
           'tS': 'open_long',
           'tgtCcy': 'USDT',
           'triggerPx': '55.000000000',
           'triggerPxType': 'mark',
           'triggerTime': '1684060542627',
           'userId': '3704614084',
           'version': 1041590898159853568},
  'lastTradeTimestamp': None,
  'postOnly': None,
  'price': 50.0,
  'reduceOnly': None,
  'remaining': None,
  'side': 'buy',
  'status': 'open',
  'stopPrice': 55.0,
  'symbol': 'LTC/USDT:USDT',
  'timeInForce': None,
  'timestamp': 1684060542627,
  'trades': [],
  'triggerPrice': 55.0,
  'type': 'limit'}]
```

**Review by carlosmiei — APPROVED** [review]

**Comment by dmunozfer** [comment]
Thanks! 👍

**Comment by dmunozfer** [comment]
@carlosmiei It seems that the bitget.js file has not been updated and has the old version that does not allow to subscribe to ordersAlgo. What could be the problem?

**Comment by carlosmiei** [comment]
> @carlosmiei It seems that the bitget.js file has not been updated and has the old version that does not allow to subscribe to ordersAlgo. What could be the problem?

@dmunozfer the JS version seems ok: https://github.com/ccxt/ccxt/blob/master/js/src/pro/bitget.js
If you're referring to the npm package, you have to update it to 3.0.103+ to get the update

**Pull request merged**